tests/filecheck/dialects/scf/unregistered.mlir

```diff
@@ -0,0 +1,22 @@
+// RUN: xdsl-opt %s --print-op-generic --allow-unregistered-dialect | xdsl-opt --allow-unregistered-dialect | filecheck %s
+
+// CHECK:      func.func @for_unregistered() {
+// CHECK-NEXT:   %lb = arith.constant 0 : index
+// CHECK-NEXT:   %ub = arith.constant 42 : index
+// CHECK-NEXT:   %s = arith.constant 3 : index
+// CHECK-NEXT:   scf.for %iv = %lb to %ub step %s {
+// CHECK-NEXT:     "unregistered_op"() : () -> ()
+// CHECK-NEXT:   }
+// CHECK-NEXT:   func.return
+// CHECK-NEXT: }
+
+func.func @for_unregistered() {
+  %lb = arith.constant 0 : index
+  %ub = arith.constant 42 : index
+  %s = arith.constant 3 : index
+  scf.for %iv = %lb to %ub step %s {
+    "unregistered_op"() : () -> ()
+    scf.yield
```

Comment:
Am I missing something?
```suggestion
    "unregistered_op"() : () -> ()
```

Reply:
You're missing that I believe this doesn't verify as the implicit terminator isn't added, due to an unrelated problem this pr was trying to solve. (`scf.for` expects to be terminated by an `scf.yield`)

Reply:
I'm confused, does this test not pass on main?

Reply:
OK I just tested it out myself and became more confused :) I think a part of the problem is that the code in the printer is a little backwards, and would benefit from a comment. It might also benefit from a bit of a refactor, it feels a little wasteful to check for terminator trait for every operation that we print, when they can only be the last operation in the block. That's probably out of scope for this PR though.